aztk/spark/helpers/job_submission.py

```diff
@@ -0,0 +1,144 @@
+#     recent_run_job = __get_recent_job(spark_client, job_id)
+#     if recent_run_job.id and recent_run_job.state == batch_models.JobState.active:
+#         spark_client.batch_client.job.disable(job_id=recent_run_job.id, disable_tasks=batch_models.DisableJobOption.requeue)
+   
+#     # disable the job_schedule
+#     spark_client.batch_client.job_schedule.disable(job_id)
+
+
+# def enable(spark_client, job_id):
+#     # disable the currently running job from the job schedule if exists
+#     recent_run_job = __get_recent_job(spark_client, job_id)
+#     if recent_run_job.id and recent_run_job.state == batch_models.JobState.active:
+#         spark_client.batch_client.job.enable(job_id=recent_run_job.id)
+   
+#     # disable the job_schedule
+#     spark_client.batch_client.job_schedule.enable(job_id)
+
+
+def stop(spark_client, job_id):
+    # terminate currently running job and tasks
+    recent_run_job = __get_recent_job(spark_client, job_id)
+    spark_client.batch_client.job.terminate(recent_run_job.id)
+    # terminate job_schedule
+    spark_client.batch_client.job_schedule.terminate(job_id)
+
+
+def delete(spark_client, job_id):
+    recent_run_job = __get_recent_job(spark_client, job_id)
+    deleted_job_or_job_schedule = False
+    # delete job
+    try:
+        spark_client.batch_client.job.delete(recent_run_job.id)
+        deleted_job_or_job_schedule = True
+    except batch_models.batch_error.BatchErrorException:
+        pass
+    # delete job_schedule
+    try:
+        spark_client.batch_client.job_schedule.delete(job_id)
+        deleted_job_or_job_schedule = True
+    except batch_models.batch_error.BatchErrorException:
+        pass
+
+    return deleted_job_or_job_schedule
+
+def get_application(spark_client, job_id, application_name):
+    # info about the app
+    recent_run_job = __get_recent_job(spark_client, job_id)
+    return spark_client.batch_client.task.get(job_id=recent_run_job.id, task_id=application_name)
+
+
+def get_application_log(spark_client, job_id, application_name):
+    # TODO: change where the logs are uploaded so they aren't overwritten on scheduled runs
+    #           current: job_id, application_name/output.log
+    #           new: job_id, recent_run_job.id/application_name/output.log
+
+    # recent_run_job = __get_recent_job(spark_client, job_id)
+    # return spark_client.get_application_log(recent_run_job.id.replace(":", "-"), application_name)
+    return spark_client.get_application_log(job_id, application_name)
+
+
+def stop_app(spark_client, job_id, application_name):
+    recent_run_job = __get_recent_job(spark_client, job_id)
+
+    # TODO: stop spark job on node 
+
+    # stop batch task
+    try:
+        spark_client.batch_client.task.terminate(job_id=recent_run_job.id, task_id=application_name)
+        return True
+    except batch_models.batch_error.BatchErrorException:
+        return False
+
+def wait_until_job_finished(spark_client, job_id):
```

Comment:
I think we've had this conversation in the past, but I really don't like methods like this built into the core SDK - it should happen at a higher level (like the CLI or calling client). There is little to no monitoring here and a caller can't control most of the polling logic.